### PR TITLE
Colors

### PR DESCRIPTION
**Pull request by dennisdolliver**
This ports the colors from https://github.com/kyndryl-digital/kyndryl-storybook

### CODE REVIEW

File: src/scss/variables/colors.scss [hunk 1]
```diff
@@ -1,23 +1,207 @@
...
+  --k-color-darkstone-50: #898888;
+  --k-color-darkstone-20: #dcdad8;
+  --k-color-darkstone-10: #f9f9f9;
+  --k-color-spruce-80: #163d43;
+  --k-color-spruce-70: #205961;
+  --k-color-spruce-60: #29707a;
+  --k-color-spruce-40: #3797a4;
+  --k-color-spruce-30: #3fadbd;
+  --k-color-warmred-50: #ff462d;
+  --k-color-warmred-40: #ff6b57;
```
Comment (brian-patrick-3) [inline src/scss/variables/colors.scss:15]: These seem like they're repeated below? Let's make sure there's no duplicates.

File: src/stories/Colors.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,893 @@
+import { Meta, Unstyled, Source } from '@storybook/blocks';
+
+<Meta title="Foundation/Colors" />
+
+[Figma Design Source](https://www.figma.com/file/tTu138pz3jLk0mykJSg3t4/KDS---Foundation-v1.0.0?type=design&node-id=965-3806&mode=design&t=SyYtt2cw0sKnToqf-0)
```
Comment (brian-patrick-3) [inline src/stories/Colors.stories.mdx:5]: Maybe move this under the "Colors" heading.

**Review by brian-patrick-3 — CHANGES_REQUESTED** [review]
On he MDX page, it seems like some of the color previews are missing, duplicated, or out of order. Could stand to clean that up a bit. Otherwise, just a couple small comments.

The design delivery seems to have a bunch of stuff related to themes now too, so we'll probably need to figure out some mixins and utility classes to support that. @dennisdolliver does your codebase have anything existing for theming currently? My thought is we basically have a class like `.kd-theme--name` and change the tokens within.

**Comment by dennisdolliver** [comment]
> On he MDX page, it seems like some of the color previews are missing, duplicated, or out of order. Could stand to clean that up a bit. Otherwise, just a couple small comments.
> 
> The design delivery seems to have a bunch of stuff related to themes now too, so we'll probably need to figure out some mixins and utility classes to support that. @dennisdolliver does your codebase have anything existing for theming currently? My thought is we basically have a class like `.kd-theme--name` and change the tokens within.

Yeah that's how it works in ours. We have a theme scss per theme available that sets the variables to the element named variables that's scoped to a class named for that theme.

.#{$prefix-class-theme}-spruce-60 {
  --kd-color-token-base: var(--k-color-spruce-60);
  --kd-color-token-inverse: var(--kd-color-brand-cloud);
  --kd-color-token-interactive: var(--kd-color-brand-cloud);
  --kd-color-token-accent: var(--kd-color-brand-cloud);
  --kd-color-eyebrow: var(--k-color-springgreen-20);
  @include css-vars-color-usage;
}

**Review by brian-patrick-3 — CHANGES_REQUESTED** [review]
Looking pretty good now overall, a couple things I think we should address still:

1. Let's change the token prefix from `--k-` to `--kd-` so everything we have is consistent.
2. We'll need to create some utility classes for applying the themes, like Dennis shared above. Idea would be to add the mixin to a utility class, add the class to the root tag (`<html>`, or would work anywhere really), and the theme variables would override the default ones. With that in mind, I think we need to name the tokens within the theme mixins the same as the base tokens.
3. Add some Usage examples to the MDX page for Mixins/Utilities like we did for Grid & Typography.

**Comment by peterpham-kyndryl** [comment]
> Looking pretty good now overall, a couple things I think we should address still:
> 
> 1. Let's change the token prefix from `--k-` to `--kd-` so everything we have is consistent.
> 2. We'll need to create some utility classes for applying the themes, like Dennis shared above. Idea would be to add the mixin to a utility class, add the class to the root tag (`<html>`, or would work anywhere really), and the theme variables would override the default ones. With that in mind, I think we need to name the tokens within the theme mixins the same as the base tokens.
> 3. Add some Usage examples to the MDX page for Mixins/Utilities like we did for Grid & Typography.
Will changing from `--k-` to `--kd-` affect your team's code, @dennisdolliver?

File: src/scss/variables/colors.scss [hunk 3]
```diff
@@ -1,23 +1,290 @@
...
+    --k-theme-color-cloud-accent-tertiary: var(--k-color-cloud-20);
+  }
+
+  @mixin light-theme-white {
+    --k-theme-color-white-background: var(--k-color-neutral-white);
+    --k-theme-color-white-foreground: var(--k-color-darkstone-80);
+    --k-theme-color-white-accent-primary: var(--k-color-warmred-50);
+    --k-theme-color-white-accent-secondary: var(--k-color-spruce-60);
+    --k-theme-color-white-accent-tertiary: var(--k-color-darkstone-10);
+  }
```
Comment (brian-patrick-3) [inline src/scss/variables/colors.scss:289]: Should move the mixins to a separate file in the mixins folder.

**Comment by brian-patrick-3** [comment]
> 2. I think we need to name the tokens within the theme mixins the same as the base tokens.
> What do you mean by this?

Ex: 

1. We have `--kd-color-background-primary` for the main background color.
2. We apply a theme by adding utility class `.kd-theme--darkstone-80` to the `html` tag.
3. In that theme mixin, it should set the same variable to a new value, because havign a different token wouldn't actually change anything.
    1. Current:  `--kd-theme-color-darkstone-background: var(--kd-color-darkstone-80);`
    2. Should be: `--kd-color-background-primary: var(--kd-color-darkstone-80);`

Does that make sense? IMO, the theme stuff hasn't been fully thought out.

File: src/scss/utility/colors.scss [hunk 4]
```diff
@@ -0,0 +1,42 @@
+@use '../mixins/colors.scss';
+
+.kd-theme-darkstone-80 {
```
Comment (brian-patrick-3) [inline src/scss/utility/colors.scss:3]: For the utility classes. let's use a double dash for the modifier. Ex: `.kd-theme--darkstone-80`

**Comment by peterpham-kyndryl** [comment]
> 2\. `html` tag.

From what I understand is that we have a same set of tokens within the them mixins so that when they switch theme, they can point this set to the new set of color values as they desire?

File: src/scss/utility/colors.scss (hunk 4 above)
Comment (peterpham-kyndryl) [reply src/scss/utility/colors.scss:3]: > For the utility classes. let's use a double dash for the modifier. Ex: `.kd-theme--darkstone-80`

Ah, the BEM standard. Ok.

**Comment by peterpham-kyndryl** [comment]
> Current: --kd-theme-color-darkstone-background: var(--kd-color-darkstone-80);
Should be: --kd-color-background-primary: var(--kd-color-darkstone-80);
These mixins have background and foreground and they don't seem to have the background-secondary. So, why does it have to be --kd-color-background-primary instead of just --kd-color-background?

File: src/scss/utility/colors.scss (hunk 4 above)
Comment (brian-patrick-3) [reply src/scss/utility/colors.scss:3]: Just include this change, then we're good to merge IMO. Follow up on themes later.

**Review by brian-patrick-3 — APPROVED** [review]

**Review by cunningham-kyndryl — APPROVED** [review]
Approved but need to make sure this has alerts colors. Also, may need to add Bridge Status' colors for use on tables and other areas as things go through workflows.

**Pull request merged**